### PR TITLE
Update to gfx hal 0.3

### PR DESCRIPTION
**Pull request by rukai**
closes https://github.com/gfx-rs/wgpu/issues/275

### CODE REVIEW

File: wgpu-native/src/conv.rs [hunk 1]
```diff
@@ -219,50 +222,46 @@ pub fn map_depth_stencil_state_descriptor(
...
-    stencil_write_mask: u32,
 ) -> hal::pso::StencilFace {
     hal::pso::StencilFace {
         fun: map_compare_function(stencil_state_face_desc.compare),
-        mask_read: hal::pso::State::Static(stencil_read_mask), // TODO dynamic?
-        mask_write: hal::pso::State::Static(stencil_write_mask), // TODO dynamic?
         op_fail: map_stencil_operation(stencil_state_face_desc.fail_op),
         op_depth_fail: map_stencil_operation(stencil_state_face_desc.depth_fail_op),
         op_pass: map_stencil_operation(stencil_state_face_desc.pass_op),
-        reference: hal::pso::State::Static(0), // TODO can this be set?
```
Comment (rukai) [inline wgpu-native/src/conv.rs]: No idea what these are for, but they are gone now? I just deleted them.

File: wgpu-native/src/conv.rs [hunk 2]
```diff
@@ -219,50 +222,46 @@ pub fn map_depth_stencil_state_descriptor(
...
-            hal::pso::StencilTest::On {
-                front: map_stencil_face(
-                    &desc.stencil_front,
-                    desc.stencil_read_mask,
-                    desc.stencil_write_mask,
-                ),
-                back: map_stencil_face(
-                    &desc.stencil_back,
-                    desc.stencil_read_mask,
-                    desc.stencil_write_mask,
```
Comment (rukai) [inline wgpu-native/src/conv.rs]: Looks like the values gone from StencilFace popped up here..?
Should I be setting these values to the same values that were previously used in StencilFace?

File: wgpu-native/src/conv.rs (hunk 1 above)
Comment (kvark) [reply wgpu-native/src/conv.rs]: They are at the level above now: `reference_values` and such

File: ffi/wgpu.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #define WGPU_LOCAL
 
-/* Generated with cbindgen:0.9.0 */
+/* Generated with cbindgen:0.8.6 */
```
Comment (kvark) [inline ffi/wgpu.h]: gotta update your cbindgen? :)

File: wgpu-native/Cargo.toml [hunk 4]
```diff
@@ -28,16 +28,16 @@ bitflags = "1.0"
...
 log = "0.4"
-hal = { package = "gfx-hal", version = "0.2.1" }
-gfx-backend-empty = { version = "0.2.1" }
-gfx-backend-vulkan = { version = "0.2", optional = true }
-gfx-backend-dx11 = { version = "0.2.1", optional = true }
-gfx-backend-dx12 = { version = "0.2.1", optional = true }
-gfx-backend-metal = { version = "0.2.2", optional = true }
-gfx-backend-gl = { version = "0.2", optional = true }
+hal = { package = "gfx-hal", version = "0.3.0" }
+gfx-backend-empty = { version = "0.3.0", features = ["winit"] }
```
Comment (kvark) [inline wgpu-native/Cargo.toml]: `winit` feature isn't the default, we only need it enabled when something like `window-winit` feature is enabled here

File: wgpu-native/src/conv.rs (hunk 2 above)
Comment (kvark) [reply wgpu-native/src/conv.rs]: I think what you are doing right now (just setting the stencil to `Dynamic`) is correct

File: wgpu-native/src/conv.rs (hunk 1 above)
Comment (kvark) [reply wgpu-native/src/conv.rs]: correct

File: wgpu-native/Cargo.toml (hunk 4 above)
Comment (rukai) [reply wgpu-native/Cargo.toml]: I tried doing: ```window-winit = ["winit", "gfx-backend-empty/winit", "gfx-backend-vulkan/winit", "gfx-backend-dx11/winit", "gfx-backend-dx12/winit", "gfx-backend-metal/winit", "gfx-backend-gl/winit", "gfx-backend-gl/glutin"]```
But that doesn't work as it now tries to compile every backend.

Comment (rukai) [reply wgpu-native/Cargo.toml]: Unless you can explain how this is supposed to work, I'll leave it for someone else to investigate. I have not regressed any functionality here, the optional winit feature is new gfx hal 0.3

**Comment by kvark** [comment]
Fair enough!
Bors r+

**Pull request merged**